### PR TITLE
Write Z-Velocity and temperature only when solved

### PR DESCRIPTION
**Pull request by bigfooted**
## Proposed Changes
 
Do not write Z-velocity or temperature when the (adjoint) equations are not solved. 

## PR Checklist

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [x] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: TestCases/serial_regression_AD.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def main():
     discadj_cht.cfg_dir   = "coupled_cht/disc_adj_incomp_2d"
     discadj_cht.cfg_file  = "cht_2d_3cylinders.cfg"
     discadj_cht.test_iter = 10
-    discadj_cht.test_vals = [-2.364405, -3.085551, -3.085518, -3.085513] #last 4 columns
```
[the remark this answers is not in the document]
Comment (bigfooted) [reply TestCases/serial_regression_AD.py:227]: The BGS residuals are used here, and this one is the average over the solution variables, which now excludes the  Z-velocity.

File: SU2_CFD/src/output/CAdjFlowIncOutput.cpp [hunk 2]
```diff
@@ -109,9 +109,9 @@ void CAdjFlowIncOutput::SetHistoryOutputFields(CConfig *config){
   /// DESCRIPTION: Root-mean square residual of the adjoint Velocity y-component.
   AddHistoryOutput("RMS_ADJ_VELOCITY-Y", "rms[A_V]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint Velocity y-component.", HistoryFieldType::RESIDUAL);
   /// DESCRIPTION: Root-mean square residual of the adjoint Velocity z-component.
-  AddHistoryOutput("RMS_ADJ_VELOCITY-Z", "rms[A_W]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint Velocity z-component.", HistoryFieldType::RESIDUAL);
+  if (nDim == 3) AddHistoryOutput("RMS_ADJ_VELOCITY-Z", "rms[A_W]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint Velocity z-component.", HistoryFieldType::RESIDUAL);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CAdjFlowIncOutput.cpp:112]:
```suggestion
  if (nDim == 3) {
    AddHistoryOutput("RMS_ADJ_VELOCITY-Z", "rms[A_W]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint Velocity z-component.", HistoryFieldType::RESIDUAL);
  }
```

File: SU2_CFD/src/output/CAdjFlowIncOutput.cpp [hunk 3]
```diff
@@ -109,9 +109,9 @@ void CAdjFlowIncOutput::SetHistoryOutputFields(CConfig *config){
...
   /// DESCRIPTION: Maximum residual of the temperature.
-  AddHistoryOutput("RMS_ADJ_TEMPERATURE", "rms[A_T]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint temperature.", HistoryFieldType::RESIDUAL);
+  if (heat || weakly_coupled_heat) AddHistoryOutput("RMS_ADJ_TEMPERATURE", "rms[A_T]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint temperature.", HistoryFieldType::RESIDUAL);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CAdjFlowIncOutput.cpp:114]:
```suggestion
  if (heat || weakly_coupled_heat) {
    AddHistoryOutput("RMS_ADJ_TEMPERATURE", "rms[A_T]", ScreenOutputFormat::FIXED, "RMS_RES", "Root-mean square residual of the adjoint temperature.", HistoryFieldType::RESIDUAL);
  }
```

File: SU2_CFD/src/output/CAdjFlowIncOutput.cpp [hunk 4]
```diff
@@ -133,9 +133,9 @@ void CAdjFlowIncOutput::SetHistoryOutputFields(CConfig *config){
   /// DESCRIPTION: Maximum residual of the adjoint Velocity y-component
   AddHistoryOutput("MAX_ADJ_VELOCITY-Y", "max[A_RhoV]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the adjoint Velocity y-component", HistoryFieldType::RESIDUAL);
   /// DESCRIPTION: Maximum residual of the adjoint Velocity z-component
-  AddHistoryOutput("MAX_ADJ_VELOCITY-Z", "max[A_RhoW]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the adjoint Velocity z-component", HistoryFieldType::RESIDUAL);
+  if (nDim == 3) AddHistoryOutput("MAX_ADJ_VELOCITY-Z", "max[A_RhoW]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the adjoint Velocity z-component", HistoryFieldType::RESIDUAL);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CAdjFlowIncOutput.cpp:136]:
```suggestion
  if (nDim == 3) {
    AddHistoryOutput("MAX_ADJ_VELOCITY-Z", "max[A_RhoW]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the adjoint Velocity z-component", HistoryFieldType::RESIDUAL);
  }
```

File: SU2_CFD/src/output/CAdjFlowIncOutput.cpp [hunk 5]
```diff
@@ -133,9 +133,9 @@ void CAdjFlowIncOutput::SetHistoryOutputFields(CConfig *config){
...
   /// DESCRIPTION: Maximum residual of the temperature.
-  AddHistoryOutput("MAX_ADJ_TEMPERATURE", "max[A_T]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the temperature.", HistoryFieldType::RESIDUAL);
+  if (heat || weakly_coupled_heat) AddHistoryOutput("MAX_ADJ_TEMPERATURE", "max[A_T]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the temperature.", HistoryFieldType::RESIDUAL);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CAdjFlowIncOutput.cpp:138]:
```suggestion
  if (heat || weakly_coupled_heat) {
    AddHistoryOutput("MAX_ADJ_TEMPERATURE", "max[A_T]", ScreenOutputFormat::FIXED, "MAX_RES", "Maximum residual of the temperature.", HistoryFieldType::RESIDUAL);
  }
```

File: SU2_CFD/src/output/CAdjFlowIncOutput.cpp [hunk 6]
```diff
@@ -148,9 +148,9 @@ void CAdjFlowIncOutput::SetHistoryOutputFields(CConfig *config){
   /// DESCRIPTION: BGS residual of the adjoint Velocity y-component
   AddHistoryOutput("BGS_ADJ_VELOCITY-Y", "bgs[A_RhoV]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint Velocity y-component", HistoryFieldType::RESIDUAL);
   /// DESCRIPTION: BGS residual of the adjoint Velocity z-component
-  AddHistoryOutput("BGS_ADJ_VELOCITY-Z", "bgs[A_RhoW]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint Velocity z-component", HistoryFieldType::RESIDUAL);
+  if (nDim == 3) AddHistoryOutput("BGS_ADJ_VELOCITY-Z", "bgs[A_RhoW]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint Velocity z-component", HistoryFieldType::RESIDUAL);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CAdjFlowIncOutput.cpp:151]:
```suggestion
  if (nDim == 3) {
    AddHistoryOutput("BGS_ADJ_VELOCITY-Z", "bgs[A_RhoW]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint Velocity z-component", HistoryFieldType::RESIDUAL);
  }
```

File: SU2_CFD/src/output/CAdjFlowIncOutput.cpp [hunk 7]
```diff
@@ -148,9 +148,9 @@ void CAdjFlowIncOutput::SetHistoryOutputFields(CConfig *config){
...
   /// DESCRIPTION: BGS residual of the temperature.
-  AddHistoryOutput("BGS_ADJ_TEMPERATURE", "bgs[A_T]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint temperature.", HistoryFieldType::RESIDUAL);
+  if (heat || weakly_coupled_heat) AddHistoryOutput("BGS_ADJ_TEMPERATURE", "bgs[A_T]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint temperature.", HistoryFieldType::RESIDUAL);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CAdjFlowIncOutput.cpp:153]:
```suggestion
  if (heat || weakly_coupled_heat) {
    AddHistoryOutput("BGS_ADJ_TEMPERATURE", "bgs[A_T]", ScreenOutputFormat::FIXED, "BGS_RES", "BGS residual of the adjoint temperature.", HistoryFieldType::RESIDUAL);
  }
```

**Review by pcarruscag — COMMENTED** [review]
The compressible output needs the same change

**Review by pcarruscag — APPROVED** [review]

**Pull request merged**